linera-views/src/batch.rs

```diff
@@ -373,63 +373,60 @@ impl DeletePrefixExpander for MemoryContext<()> {
     }
 }
 
-/// The simplified batch used for the computation.
+/// A simplified batch used for the computation.
 #[async_trait]
 pub trait SimplifiedBatch: Sized + Send + Sync {
-    /// The Iterator type used for the batch
-    type Iter: SimplifiedBatchIter;
+    /// The iterator type used to process values from the batch.
+    type Iter: BatchValueWriter<Self>;
 
-    /// Returns an owning iterator on the simplified batch.
+    /// Returns an owning iterator over the values in the batch.
     fn into_iter(self) -> Self::Iter;
 
-    /// Returns the total number of entries of the simplified batch
+    /// Returns the total number of entries in the batch.
     fn len(&self) -> usize;
 
-    /// Returns the total number of bytes of the simplified batch
-    fn bytes(&self) -> usize;
+    /// Returns the total number of bytes in the batch.
+    fn num_bytes(&self) -> usize;
 
-    /// Returns the overhead size of the simplified batch
+    /// Returns the overhead size of the batch.
     fn overhead_size(&self) -> usize;
 
-    /// Appends the iterator to the simplified batch
-    fn try_append(
-        &mut self,
-        iter: &mut Self::Iter,
-        value_size: &mut usize,
-    ) -> Result<bool, bcs::Error>;
-
-    /// Returns the serialization and clears the simplified batch
-    fn to_bytes(&mut self) -> Result<Vec<u8>, bcs::Error>;
-
-    /// Adds an individual delete operation
+    /// Adds an individual delete operation.
     fn add_delete(&mut self, key: Vec<u8>);
 
-    /// Adds an individual insert of (key,value)
+    /// Adds an individual insert of (key,value).
```

Comment:
```suggestion
    /// Adds an individual insert of `(key, value)`.
```

(Or "insertion of a key-value pair"?)
(And maybe the "individual" is unnecessary, also above.)